### ISSUE TITLE
[Feature]: Required form fields should have an indicator

### ISSUE THREAD

**Issue by momcilo-appsmith** [appsmithorg/appsmith#10905]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

People usually go through forms looking for the required fields, if there are no indicators for them then they might fill everything out even if it is not required - that might make their process longer than needed.

### Why should this be worked on?

It will help our developers reduce the time needed to fill out long forms in their apps.

**Comment by dilippitchika** [comment]
@momcilo-appsmith will discuss, but thinking that if there's a label we should add a * at the end of the label. I think this is a small issue which we should just resolve.

It will not solve for all cases, but works for most.

**Comment by Nikhil-Nandagopal** [comment]
I don't believe there is a standard way to communicate this and it should be a choice of the user so if they want to add *, they will, if they want to write (required) they will. I think we should wash our hands off this and let the user decide since there is no clear industry standard we can be sure users will be happy with.
@dilippitchika @momcilo-appsmith

**Comment by dilippitchika** [comment]
@Nikhil-Nandagopal in this case the problem is the end-user suffers because we are making it hard for our devs to add something which is very common for forms. 

If so, should we add a setting in the form to "add *" or "required" label for all required fields? Users can still customise on their own but in this scenario, we are not forcing them but just suggesting.

**Comment by Nikhil-Nandagopal** [comment]
@dilippitchika why is it hard? Isn't it the same as just adding * in the label field? Every setting we add clutters the property pane so we have to keep choosing these well

**Comment by dilippitchika** [comment]
More like devs forget adding */optional until later, the larger the form the more work they do to add it later. But since they are marking a field required while building, If they have used anything like google forms or other form builders, they know they don't need to worry about adding * or required themselves. 

Problem here is I want to minimize rework later on, which then leads to people asking us question about why we don't support this out-of-the-box.

**Comment by Nikhil-Nandagopal** [comment]
@dilippitchika if developers are forgetting to add the *, then doesn't it stand to reason that they might be forgetting to mark it as required as well? Maybe the solution should be to allow group editing of properties? 
We have to remember Appsmith is not just a form builder and think about all the use cases. An example is users create Login screens with Appsmith where the email is required but login screens don't generally have a * after the email? Similarly, there are cases where this would be unexpected for users and they would struggle to figure out how to get out of this state.
So to me, the problem doesn't feel severe and the solution isn't straightforward & feels like it will cause us more problems.
